### ISSUE TITLE
ensure a consist treatment of dust attenuation in the mocks and imaging surveys

### ISSUE THREAD

**Issue by moustakas** [desihub/desitarget#158]
Quoting from an off-list email from @apcooper.  Aspects of this message should also be cross-listed in the fiberassign module.
```
However, this part of the code is also related to question I raised
yesterday with Jaime regarding if/how a dust map is being applied to
the different mocks. For example the MWS code you point to:

https://github.com/desihub/desitarget/blob/master/py/desitarget/mock/selection.py#L117

an 'obs_rflux' is calculated by using an EBV value, which I think is
being assigned brick-by-brick elsewhere in desitarget (right?).
However, this obs_rflux is not used for the MWS_MAIN selection,
although it is passed to the function that selects F standards.

I think we need to double check what is happening here, and in
particular that we're handling extinction in the right way for MWS
(possibly BGS too).

In the case of the MWS, dust is already included in the SDSSr_true and
SDSSr_obs magnitudes reported in mock. Quoting from my READ_ME for
these mocks:

SDSSx_true         : apparent magnitudes in SDSS bands (x = ugriz),
including extinction.
SDSSx_true_nodust  : apparent magnitudes in SDSS bands (x = ugriz),
excluding extinction.
SDSSx_obs          : apparent magnitudes drawn from photometric error
distribution.

So the value to be used for selection is either SDSSr_obs (which
includes both dust and photometric error) or SDSSr_true (which
includes only dust). I don't know about the other mocks but in the MWS
case I think photometric error (and brick-to-brick depth variations)
are not important, so we can focus on extinction.

If galaxia and desitarget assume different dust maps, using the
_true_nodust values and applying a different dust map in desitarget
would not be the right thing to correct for that, because the galaxia
mocks are generated to a fixed depth *after* extinction.

Separately, I've seen a plot that suggests the ratio of assigned BGS
fibres to BGS targets has a strong galactic latitude dependence (lower
towards the plane) with a patchiness that makes it look a lot like the
result of dust. BGS has no 'built in' dust (i.e. the catalog is
complete to a limit in unextincted apparent magnitude).

This variation for BGS was confusing to me. I would have expected that
extinction *increases* that (targeted/available) ratio near the plane,
by eliminating faint targets from an otherwise roughly uniform
distribution. In other words, the list of targets (targets.fits) being
fed to fibre assignment should have lower density near the plane.
Because BGS is top priority for bright conditions, a greater fraction
of available BGS targets should then get fibres near the plane.

You might hear about this officially from the BGS side, or it might
turn out to be a bug -- in any case I couldn't find a ready
explanation, which suggests we should make sure that we've documented
the procedure for applying a dust map (and depth variations) in
desitarget, and that this is applied in a way that takes  account of
the differences between the individual mocks (or that the mocks are
made in a consistent way in the first place).
```

**Comment by apcooper** [comment]
@moustakas beat me to raising this here, thanks John. So to clarify:

* I don't think this issue is responsible for the fiber assignment efficiency variations seen in BGS, (raised [here](https://github.com/desihub/fiberassign/issues/37#issuecomment-286407464) for fiberassign as the continuation of an earlier issue) because my understanding of the `desitarget` code is that the extinction correction applied in `cuts.py` is _not_ currently applied to the mocks. For BGS this is correct: the magnitudes in the Durham MXXL mocks do not include any extinction to begin with, so they shouldn't be corrected for dust.

* For MWS the situation is that the mocks contain magnitudes _including_ dust extinction, and _we currently want to use these extincted magnitudes for selection_ so these shouldn't be corrected for extinction either. The correction is (wrongly?) applied in the calculation of `obs_rflux` in the MWS mock selection code, but then this is not used for anything except the standards.

I'm less worried about this now that we've concluded this isn't related to the BGS issue and I've figured out the mock selection is not being done through cuts.py but by separate mock-only code in [mocks/selection.py](https://github.com/desihub/desitarget/blob/8f2d72c2b58fdb11dfdf4960a2d90f139380f350/py/desitarget/mock/selection.py). Still there seems to be potential for confusion here.

**Comment by moustakas** [comment]
In the version of the mock selection code currently in master, I am selecting MWS_MAIN and MWS_MAIN_VERY_FAINT targets (see [here](https://github.com/desihub/desitarget/blob/master/py/desitarget/mock/selection.py#L101-L111)) by applying *extinction-corrected, DECaLS* r-band flux limits of 15<r<19 and 19<r<20, respectively.

Based on the comment above by @apcooper, this is *wrong*.  At some point in the future there will be MWS target selection criteria based on observational data (e.g., Gaia) rather than perfect knowledge of a perfect mock.  But to address this issue for now I will simply pass forward--i.e., assign the MWS_MAIN targeting bit--to *all* the targets in the MWS_MAIN/galaxia mock, which as I understand it is flux-limited to r_SDSS < 20 (including extinction).  

Getting a little further into the weeds:  Because of the way the code is structured, the attenuated (well, extincted) SDSS r-band magnitude which is in the mock is not trivial to access in this part of the code.  Therefore I will simply assign a MWS_MAIN targeting bit to everything (down to r_SDSS,obs=20) and drop the MWS_MAIN_VERY_FAINT bit for now.

As for the standard stars, I am selecting those using the extincted (i.e., observed) DECaLS r-band flux, where I have applied Galactic extinction in a way that (should be!) consistent with the imaging survey.  We should assess this selection of course but for now it should be "good enough."

@apcooper Does all this sound reasonable?  Any other comments?

**Comment by apcooper** [comment]
I think I appreciate the problem, but I don't follow the reasoning for dropping MWS_MAIN_VERY_FAINT as a separate sample.

The idea of a separate target class for 19<r<20 was to provide a dense enough set of filler targets while aiming for completeness of MWS_MAIN down to a brighter limit (r=19, leaving aside whether this is with/without dust). Since the density of the faint filler targets is higher than those of the faintest MAIN targets, we want the filler to have lower priority.

A more natural fix might be to just adopt the same logic for MWS_MAIN (and its subsets) as all the other surveys: selecting on the extinction-corrected DECaLS magnitude. That could save a lot of coding confusion and work further down the line.

The precise faint limit (either for MWS or BGS) is waiting on estimates of parameter error etc. from the spectral simulations. Maybe I’m not the best person to judge this, but in the meantime I expect a cut at r_without_extinction = 19 isn't all that different from one at r_with_extinction =19, at a whole-survey level and to the degree of approximation we're working at right now.

In terms of the impact on the number of faint stars in the MAIN survey, my hunch is that we should be more worried about dropping the distinction between MAIN stars and filler than about switching from cutting on corrected vs. uncorrected magnitudes.

The motivation for selection on r_with_extinction (if I remember correctly) was to not waste fibres on stars with large corrections, since (unlike BGS) good spectra are more useful to us than clean sampling statistics, at the faint limit of the survey. I don’t know what fraction of the footprint gives corrections high enough for this to be a big deal, but my hunch is that it’s small, and the benefits from streamlining  will outweigh the marginal loss of efficiency (which we could sort out with sub-priority tweaks in regions of very high extinction, if it’s worth it).

What do you think, @moustakas? Remaking the galaxia mock to be complete to r_SDSS_nodust=20 is trivial, so that should’t be a factor.

I'll point the MWS list at this discussion for wider input.

**Comment by segasai** [comment]
@apcooper Just my 2c. 
1) The fraction of area where extinction is significant should be tiny indeed, but I'd say it is still worth making sure that the selection makes sense there. 
2) I'd say we need two limits. One is global hard limit on non-extinction corrected magnitude which is basically ensures we have any meaningful flux in a fiber, for example r_not_extinction_corrected < 20.5
The second limit is for the actual science selection, for example  for fillers it would be 19<r_extinction_corrected_using_schlegel<20. Obviously it may turn out that in the very extincted area that we'll be incomplete in the filler box, but I think we are fine with it.

**Comment by apcooper** [comment]
Thanks @segasai. Agree in both cases. @moustakas, do you know if/where in the code targets that have faint r_not_extinction_corrected get excluded for the other parts of the survey (e.g. BGS)? Is it straightforward to do something similar for MWS? If so I think we should go ahead and include that now, r_not_extinction_corrected < 20.5 seems a good cut to start with.

**Comment by moustakas** [comment]
Thank you for the extensive discussion, as I now have a much better understanding of the MWS mocks and the needs of the MWS group.  As requested, I've restored (in #173) the distinct MWS_MAIN and MWS_MAIN_VERY_FAINT targeting bits with magnitude cuts of 15<=r<19 and 19<=r<20, respectively, where "r" is the DECaLS extinction-corrected r-band magnitude.  (Also note the position of the equals sign, about which there was BGS-related on-list discussion!)

FYI, this "observed" r-band magnitude is derived as follows (and note that this is broadly the same algorithms used for the other target classes), which may not be what the MWS group wants---let's continue to iterate as a separate issue:
 * A KD tree of @callendeprieto's stellar spectra is built using the {Teff, logg, [Fe/H]} values.
 * Each object in @apcooper's MWS mock is assigned a spectrum based on its {Teff, logg, [Fe/H]} values.  (This means that the templates are not interpolated, which is an outstanding issue we need to fix.)
 * Each spectrum is normalized to the SDSSr_true_nodust magnitude in the mock, using the appropriate SDSS r-band filter function.
 * DECaLS grz and WISE W1W2 photometry are synthesized from the spectra using the appropriate filter function.
 * The grzW1W2 fluxes are perturbed from their true values using an empirical model of the variation in point-source depth over the DECaLS footprint developed by @geordie666.  (Actually I'm incorrectly using the typical exponential galaxy depth, which is not right -- this is on my list to fix.)
 * The target selection criteria are applied to assign targeting bits.
 * And note that the noisy and noiseless fluxes are stored in the 'targets' and 'truth' output catalogs, respectively.

In other words: as long as the mock is complete to a magnitude limit that is fainter than the noisy r-band flux limit which is being applied then you should be OK.  I'm not sure this is the case, however -- we should check or perhaps consider applying the r-band flux cut to the *noiseless* r-band flux, since ultimately the MWS selection will be done using Gaia, right?

**Comment by moustakas** [comment]
I believe this issue was addressed in #173 but if not please reopen here or create a new issue.
